addons/l10n_it_edi/models/account_edi_format.py

```diff
@@ -559,14 +609,15 @@ def _import_fattura_pa(self, tree, invoice):
                 # Collect extra info from the XML that may be used by submodules to further put information on the invoice lines
                 extra_info, message_to_log = self._l10n_it_edi_get_extra_info(company, document_type, body_tree)
 
-                partner = self._l10n_it_get_partner_invoice(tree, company)
+                partner_info = buyer_seller_info[partner_role]
+                partner = self._l10n_it_get_partner_invoice(tree, company, partner_role=partner_role)
```

Comment:
and
```suggestion
                partner_info = self._l10n_it_buyer_seller_info(company_role)
                partner = self._l10n_it_get_partner_invoice(tree, company, partner_info)
```

Reply:
I changed the function to receive the info directly instead of just the role.